src/sanexml/etree.py

```diff
@@ -0,0 +1,390 @@
+        super(XMLParser, self).__init__(
+            target=ET.TreeBuilder(insert_pis=not remove_pis, insert_comments=not remove_comments))
+
+
+def remove_attribute(element: ET.Element, attributes_to_delete):
+    attributes = element.attrib
+    for attribute in list(attributes):
+        for del_attr in attributes_to_delete:
+            if del_attr.match(attribute):
+                attributes.pop(attribute)
+                break
+
+
+# def Comment(text=None):
+#     """
+#     Comment(text=None)
+#
+#         Comment element factory. This factory function creates a special element that will
+#         be serialized as an XML comment.
+#     """
+#     return ET.Comment
+
+Comment = ET.Comment
+
+
+# class CommentedTreeBuilder(ET.TreeBuilder):
+#     def comment(self, data):
+#         self.start(ET.Comment, {})
+#         self.data(data)
+#         self.end(ET.Comment)
+
+
+def dump(elem, pretty_print=True, with_tail=True):
+    """
+    dump(elem, pretty_print=True, with_tail=True)
+
+        Writes an element tree or element structure to sys.stdout. This function
+        should be used for debugging only.
+    """
+    if pretty_print:
+        ET.indent(elem)
+    ET.dump(elem=elem)
+
+
+def Element(_tag, attrib=None, nsmap=None, **_extra):
+    """
+    Element(_tag, attrib=None, nsmap=None, **_extra)
+
+        Element factory.  This function returns an object implementing the
+        Element interface.
+
+        Also look at the `_Element.makeelement()` and
+        `_BaseParser.makeelement()` methods, which provide a faster way to
+        create an Element within a specific document or parser context.
+    """
+    if nsmap:
+        if type(nsmap) == dict:
+            for key in nsmap:
+                ET.register_namespace(key, nsmap[key])
+        else:
+            raise TypeError("nsmap should be type dictionary")
+    element = ET.Element(_tag, attrib or {}, **_extra)
+    return element
+
+
+def ElementTree(element=None, file=None, parser=None):
+    """
+    ElementTree(element=None, file=None, parser=None)
+
+        ElementTree wrapper class.
+    """
+    if element is not None:
+        return ET.ElementTree(element)
+    elif file is not None:
+        return ET.ElementTree().parse(source=file, parser=parser)
+    else:
+        return None
+
+
+def pre_process(html_content):
+    tags = set(re.findall(r'<\/?([a-zA-Z_][\w.-]*)>', html_content))
+    mapping = {tag: f"TAG{index}" for index, tag in enumerate(tags)}
+
+    def replacement(m):
+        return f"{m.group(1)}{mapping[m.group(2)]}{m.group(3)}"
+
+    processed = re.sub(r'(<\/?)([a-zA-Z_][\w.-]*)(>)', replacement, html_content)
+    return processed, mapping
+
+
+def post_process(soup, mapping):
+    reverse_mapping = {v: k for k, v in mapping.items()}
+    # print(reverse_mapping, soup)
+    for tag in soup.find_all(True):
+        if tag.name.upper() in reverse_mapping:

```

Comment:
Just to avoid calling `tag.name.upper()` twice:
```suggestion
        tag_name_upper = tag.name.upper()
        if tag_name_upper in reverse_mapping:
            tag.name = reverse_mapping[tag_name_upper]
```